### PR TITLE
Lesson10 solution

### PR DESCRIPTION
**Pull request by Ognessa**

### CODE REVIEW

File: app/src/main/java/com/onix/internship/data/storage/SensorProvider.kt [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+            type ?: "Sensor",
+            currentSubType,
+            createRandomValue(currentSubType)
+        )
+    }
+
+    private fun createRandomValue(subType: SensorSubType): String {
+        val value = when (subType) {
+            SensorSubType.SWITCH -> {
+                if (Random.nextInt(0, 2) == 0) "OFF"
```
Comment (shadwork) [inline app/src/main/java/com/onix/internship/data/storage/SensorProvider.kt:21]: should be boolean

File: app/src/main/java/com/onix/internship/network/NetworkFactory.kt [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+class NetworkFactory {
+
+    fun <S> createService(protocol: Class<S>): S {
+        return retrofit.create(protocol)
+    }
+
+    @OptIn(ExperimentalSerializationApi::class)
+    private val retrofit: Retrofit
+        get() = Retrofit.Builder()
+            .baseUrl("https://onix-systems.github.io")
```
Comment (shadwork) [inline app/src/main/java/com/onix/internship/network/NetworkFactory.kt:19]: should be constant

**Review by shadwork — CHANGES_REQUESTED** [review]
нужно немножко дописать

File: app/src/main/java/com/onix/internship/network/NetworkFactory.kt (hunk 2 above)
Comment (Ognessa) [reply app/src/main/java/com/onix/internship/network/NetworkFactory.kt:19]: Would it correct if I move this url to local properties?

Comment (shadwork) [reply app/src/main/java/com/onix/internship/network/NetworkFactory.kt:19]: No. Too hard to fiend in sources. Just create a object inside network repository

File: app/src/main/java/com/onix/internship/data/storage/SensorProvider.kt (hunk 1 above)
Comment (Ognessa) [reply app/src/main/java/com/onix/internship/data/storage/SensorProvider.kt]: Done

File: app/src/main/java/com/onix/internship/network/NetworkFactory.kt (hunk 2 above)
Comment (Ognessa) [reply app/src/main/java/com/onix/internship/network/NetworkFactory.kt]: Done

**Pull request merged**